### PR TITLE
Add table aws_kinesisanalyticsv2_application. Closes #218

### LINKED ISSUE #218: Add table aws_kinesisanalyticsv2_application
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.37.24/service/kinesisanalyticsv2

### PR DESCRIPTION
**Pull request by visiit**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
vishalkumar@Vishals-MacBook-Air aws-test % TURBOT_PROFILE=shaktiman TF_VAR_aws_profile=shaktiman ./tint.js tests/aws_kinesisanalyticsv2_application
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined
SETUP: tests/aws_kinesisanalyticsv2_application []
PRETEST: tests/aws_kinesisanalyticsv2_application
TEST: tests/aws_kinesisanalyticsv2_application
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
aws_iam_role.named_test_resource: Creating...
aws_iam_role.named_test_resource: Still creating... [10s elapsed]
aws_iam_role.named_test_resource: Creation complete after 11s [id=turbottest46291]
aws_kinesisanalyticsv2_application.named_test_resource: Creating...
aws_kinesisanalyticsv2_application.named_test_resource: Creation complete after 8s [id=arn:aws:kinesisanalytics:us-east-1:986325076436:application/turbottest46291]
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
Outputs:
account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:kinesisanalytics:us-east-1:986325076436:application/turbottest46291
resource_creation_time = 2021-04-28T14:00:38Z
resource_id = arn:aws:kinesisanalytics:us-east-1:986325076436:application/turbottest46291
resource_name = turbottest46291
version = 1
Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:kinesisanalytics:us-east-1:986323036436:application/turbottest46291"
    ],
    "application_name": "turbottest46291",
    "application_version_id": 1,
    "partition": "aws",
    "runtime_environment": "SQL-1_0",
    "service_execution_role": "arn:aws:iam::986323036436:role/turbottest46291",
    "title": "turbottest46291"
  }
]
✔ PASSED
Running SQL query: test-hydrate-query.sql
[
  {
    "application_description": null,
    "cloud_watch_logging_option_descriptions": null,
    "service_execution_role": "arn:aws:iam::986323036436:role/turbottest46291",
    "tags_src": [
      {
        "Key": "Name",
        "Value": "turbottest46291"
      }
    ]
  }
]
✔ PASSED
Running SQL query: test-list-query.sql
[
  {
    "application_arn": "arn:aws:kinesisanalytics:us-east-1:986323036436:application/turbottest46291",
    "application_name": "turbottest46291",
    "application_status": "READY",
    "application_version_id": 1
  }
]
✔ PASSED
Running SQL query: test-notfound-query.sql
null
✔ PASSED
Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:kinesisanalytics:us-east-1:986323036436:application/turbottest46291"
    ],
    "tags": {
      "Name": "turbottest46291"
    },
    "title": "turbottest46291"
  }
]
✔ PASSED
POSTTEST: tests/aws_kinesisanalyticsv2_application
TEARDOWN: tests/aws_kinesisanalyticsv2_application
SUMMARY:
1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
- Basic info

select
  application_name,
  application_arn,
  application_version_id,
  application_status,
  application_description,
  service_execution_role,
  runtime_environment
from
  aws_kinesisanalyticsv2_application;

+----------------------------+----------------------------------------------------------------------------------------+------------------------+--------------------+-------------------------+------------------------+---------------------+
| application_name           | application_arn                                                                        | application_version_id | application_status | application_description | service_execution_role | runtime_environment |
+----------------------------+----------------------------------------------------------------------------------------+------------------------+--------------------+-------------------------+------------------------+---------------------+
| test-steampipe-application | arn:aws:kinesisanalytics:us-east-1:986325076436:application/test-steampipe-application | 2                      | RUNNING            | testing steampipe query | <null>                 | SQL-1_0             |
+----------------------------+----------------------------------------------------------------------------------------+------------------------+--------------------+-------------------------+------------------------+---------------------+

- List updated application

select
  application_name,
  application_version_id,
  application_arn,
  application_status
from
  aws_kinesisanalyticsv2_application
where
  application_version_id > 1;

+----------------------------+------------------------+----------------------------------------------------------------------------------------+--------------------+
| application_name           | application_version_id | application_arn                                                                        | application_status |
+----------------------------+------------------------+----------------------------------------------------------------------------------------+--------------------+
| test-steampipe-application | 2                      | arn:aws:kinesisanalytics:us-east-1:986325076436:application/test-steampipe-application | RUNNING            |
+----------------------------+------------------------+----------------------------------------------------------------------------------------+--------------------+

- List application runtime environment as SQL

select
  application_name,
  runtime_environment,
  application_arn,
  application_status
from
  aws_kinesisanalyticsv2_application
where
  runtime_environment = 'SQL-1_0';

+----------------------------+---------------------+----------------------------------------------------------------------------------------+--------------------+
| application_name           | runtime_environment | application_arn                                                                        | application_status |
+----------------------------+---------------------+----------------------------------------------------------------------------------------+--------------------+
| test-steampipe-application | SQL-1_0             | arn:aws:kinesisanalytics:us-east-1:986325076436:application/test-steampipe-application | RUNNING            |
+----------------------------+---------------------+----------------------------------------------------------------------------------------+--------------------+

```
</details>

### CODE REVIEW

File: aws/plugin.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_inspector_assessment_target":        tableAwsInspectorAssessmentTarget(ctx),
 			"aws_inspector_assessment_template":      tableAwsInspectorAssessmentTemplate(ctx),
 			"aws_kinesis_consumer":                   tableAwsKinesisConsumer(ctx),
+			"aws_kinesisanalyticsv2_application":     tableAwsKinesisAnalyticsV2Application(ctx),
```
Comment (Subhajit97) [inline aws/plugin.go:111]: Sort the list

File: aws/service.go [hunk 2]
```diff
@@ -700,6 +701,27 @@ func KinesisService(ctx context.Context, d *plugin.QueryData, region string) (*k
 	return svc, nil
 }
 
+// KinesisAnalyticsV2Service returns the service connection for AWS Kinesis Analytics V2 service
```
Comment (Subhajit97) [inline aws/service.go:704]: Is `Kinesis Analytics V2` is a separate word, or it should be `Kinesis Analyticsv2`?

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+			},
+			{
+				Name:        "application_status",
+				Description: "The status of the application.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "create_timestamp",
+				Description: "The current timestamp when the application was created.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:53]:
```suggestion
				Type:        proto.ColumnType_TIMESTAMP,
```

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+			{
+				Name:        "application_description",
+				Description: "The description of the application.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getKinesisAnalyticsV2Application,
+			},
+			{
+				Name:        "last_update_timestamp",
+				Description: "The current timestamp when the application was last updated.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:65]:
```suggestion
				Type:        proto.ColumnType_TIMESTAMP,
```

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+//// TRANSFORM FUNCTIONS
+
+func kinesisAnalyticsV2ApplicationTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("route53resolverTagListToTurbotTags")
+
+	if d.Value == nil {
+		return nil, nil
+	}
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:228]: Do we need this block?

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+		return nil, err
+	}
+
+	return op, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func kinesisAnalyticsV2ApplicationTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("route53resolverTagListToTurbotTags")
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:224]:
```suggestion
	plugin.Logger(ctx).Trace("kinesisAnalyticsV2ApplicationTagListToTurbotTags")
```

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+
+	if d.Value == nil {
+		return nil, nil
+	}
+	tagList := d.Value.([]*kinesisanalyticsv2.Tag)
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:229]:
```suggestion
	tagList := d.HydrateItem.([]*kinesisanalyticsv2.Tag)
```

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:230]:
```suggestion

if tagList == nil {
  return nil, nil
}

```

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+			turbotTagsMap[*i.Key] = *i.Value
+		}
+	} else {
+		return nil, nil
+	}
+
+	return turbotTagsMap, nil
+}
+
+func getApplicationArn(item interface{}) *string {
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:245]: Don't use `get` prefix in function name other than hydrate functions.
```suggestion
func applicationArn(item interface{}) *string {
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@visiit There are few changes required. Please see comments, thanks!

File: aws-test/tests/aws_kinesisanalyticsv2_application/test-get-expected.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "account_id": "{{ output.account_id.value }}",
+    "akas": ["{{ output.resource_aka.value }}"],
+    "application_name": "{{ resourceName }}",
+    "application_version_id": 1,
+    "partition": "{{ output.aws_partition.value }}",
+    "runtime_environment": "SQL-1_0",
+    "service_execution_role": "arn:aws:iam::986325076436:role/{{ resourceName }}",
```
Comment (Subhajit97) [inline aws-test/tests/aws_kinesisanalyticsv2_application/test-get-expected.json:9]: Don't hardcode the account id
```suggestion
    "service_execution_role": "arn:aws:iam::{{ output.account_id.value }}:role/{{ resourceName }}",
```

File: aws-test/tests/aws_kinesisanalyticsv2_application/test-hydrate-expected.json [hunk 11]
```diff
@@ -0,0 +1,13 @@
+[
+  {
+    "application_description": null,
+    "cloud_watch_logging_option_descriptions": null,
+    "service_execution_role": "arn:aws:iam::986325076436:role/{{ resourceName }}",
```
Comment (Subhajit97) [inline aws-test/tests/aws_kinesisanalyticsv2_application/test-hydrate-expected.json:5]: Don't hardcode the account id

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+			},
+			{
+				Name:        "tags_src",
+				Description: "The key-value tags assigned to the application.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getKinesisAnalyticsV2ApplicationTags,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Standard columns for all tables
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:99]:
```suggestion
			// Steampipe standard columns
```

File: aws/table_aws_kinesisanalyticsv2_application.go [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+	// Mapping the resource tags inside turbotTags
+	var turbotTagsMap map[string]string
+	if tagList != nil {
+		turbotTagsMap = map[string]string{}
+		for _, i := range tagList {
+			turbotTagsMap[*i.Key] = *i.Value
+		}
+	} else {
+		return nil, nil
+	}
```
Comment (Subhajit97) [inline aws/table_aws_kinesisanalyticsv2_application.go:236]: Can we update this block to add check for nil first?

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@visiit There are few more changes required, otherwise it looks good. Please see comments, thanks!

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**